### ISSUE TITLE
Introduce Debug.Log filter for development

### ISSUE THREAD

**Issue by JaXt0r** [GothicVRProject/GothicVR#309]
As more and more subsystems (like world loading, animation, etc.) benefit from logs, it clutters our Debug view. The goal is therefore to introduce a system where we filter unwanted logs during development (e.g. via FeatureFlags) and therefore focus on our logs of interest.

Idea:
Debug.Log(Feature.WorldMesh, "some message");

This will be either logged or ignored based on settings at startup.

Implementation ideas:
* Unity offers ILogHandler interface to be overwritten. Context parameter could be reused etc. https://gist.github.com/nomnomab/e57d09d2319d4ee64fa315502943d54f
* log4net?
* other custom solutions from gists like this one: https://github.com/sschmid/Sherlog

**Comment by JaXt0r** [comment]
We can overwrite or extend the log functionality. Problem is, that we always get a misaligned Console view.
![image](https://github.com/GothicVRProject/GothicVR/assets/120568393/8d5bc2a6-f744-4ecb-84f0-c1744d735ece)

1. If we call an own log method and call the Unity logger afterwards, we get a totally misaligned Context (Console says, we didn't log on a class like Bootstrapper, but in a DebugLogHandler) --> If we click on it, we won't get the actual class, but the logger implementation.

```c#
public static void Log(string message, LogModule module)
        {
            if (!IsModuleEnabled(module))
                return;

            _gvrLogHandler.defaultLogHandler.LogFormat(LogType.Log, null, "[{0}] {1}", module, message);
        }
```

2. If we just wrap Unity logging behaviour, we will get a similar outcome, but at least clicking on the Console will lead us to the log creating class.

```c#
public class GvrDebugLogHandler : ILogHandler
    {
        public ILogHandler defaultLogHandler;

        public GvrDebugLogHandler(ILogHandler defaultLogHandler)
        {
            this.defaultLogHandler = defaultLogHandler;
        }

        public void LogFormat(LogType logType, string message, GvrDebug.LogModule module)
        {
            message = string.Join("[", module , "] ", message);

            defaultLogHandler.LogFormat(logType, null, message, null);
        }
```

**Comment by JaXt0r** [comment]
As Unity uses its own console with the "click" and "display" magic, we would need to leverage some new console solution at all. e.g. https://github.com/bbbscarter/UberLogger
![image](https://github.com/GothicVRProject/GothicVR/assets/120568393/2750e8b0-c1d8-4074-9a55-ec7408040b46)

**Comment by JaXt0r** [comment]
Therefore I recommend we are just cautious:
1. If we have a one-time log message like "GVR bootstrapped" it's fine to have it
2. If we have reoccurring messages, we should disable these logs via FeatureFlags and an IF-statement (or just remove the log entries from code if we don't need them any longer)

TL;DR: Let's do it as before. :-D
